spec/prometheus/client/push_spec.rb

```diff
@@ -21,6 +21,22 @@
       expect(push.gateway).to eql('http://pu.sh:1234')
     end
 
+    it 'allows to specify a custom Pushgateway with custom headers' do
+      headers = { 'Key' => 'Value' }
+      push = Prometheus::Client::Push.new('test-job', nil, nil, headers)
+
+      merged_headers = described_class::DEFAULT_HEADERS.merge(headers)
+      expect(push.headers).to eql(merged_headers)
+    end
+
+    it 'allows to averride default headers' do
```

Comment:
Actually, there is no real reason to use a different content-type.
I just showed such ability with this test case...